gslib/command.py

```diff
@@ -725,27 +729,47 @@ def RaiseInvalidArgumentException(self):
                (self.command_spec.usage_synopsis, self.command_name))
     raise CommandException(message)
 
-  def ParseSubOpts(self, check_args=False):
+  def ParseSubOpts(self,
+                   check_args=False,
+                   args=None,
+                   update_sub_opts_and_args=True):
     """Parses sub-opt args.
 
     Args:
       check_args: True to have CheckArguments() called after parsing.
-
-    Populates:
-      (self.sub_opts, self.args) from parsing.
+      args: List of args. If None, self.args will be used.
+      update_sub_opts_and_args: True if self.sub_opts and self.args should
+        be updated with the values returned after parsing. Else return a 
+        tuple of sub_opts, args returned by getopt.getopt. This is done
+        to allow this method to be called from GetGcloudStorageArgs in which
+        case we do not want to update self.sub_opts and self.args.
 
     Raises: RaiseInvalidArgumentException if invalid args specified.
     """
+    if args is None:
+      unparsed_args = self.args
+    else:
+      unparsed_args = args
     try:
-      self.sub_opts, self.args = getopt.getopt(
-          self.args, self.command_spec.supported_sub_args,
+      parsed_sub_opts, parsed_args = getopt.getopt(
+          unparsed_args, self.command_spec.supported_sub_args,
           self.command_spec.supported_private_args or [])
     except getopt.GetoptError:
       self.RaiseInvalidArgumentException()
-    if check_args:
-      self.CheckArguments()
-
-  def CheckArguments(self):
+    if update_sub_opts_and_args:
+      self.sub_opts, self.args = parsed_sub_opts, parsed_args
+      if check_args:
+        self.CheckArguments()
+    else:
+      if check_args:
+        # This is just for sanity check. Only GetGcloudStorageArgs will
+        # call this method with update_sub_opts_and_args=False and it
+        # does not set set check_args to True.
+        raise TypeError(
+            'Cannot check arguments if sub_opts and args are not updated.')
```

Comment:
Should this be "No reason to check arguments..." b/c it seems like you "can" technically do it?

Reply:
Reworded this.